cmd/tink-worker/cmd/cmd.go

```diff
@@ -0,0 +1,115 @@
+	RegistryUsername string
+	RegistryPassword string
+	WorkerID         string
+	MaxFileSize      int64
+	MaxRetry         int
+	RetryInterval    int
+	Timeout          int
+	Version          bool
+}
+
+// usage help screen.
+func Usage() {
+	usageHelp := fmt.Sprintf(`Usage:
+   tink-worker [flags]
+Flags:
+-r, --docker-registry string     Sets the Docker registry url (DOCKER_REGISTRY)
+-h, --help                       help for tink-worker
+-i, --id string                  Sets the worker id (ID)
+    --max-file-size int          Maximum file size (bytes) (MAX_FILE_SIZE) (default %d)
+    --max-retry int              Maximum number of retries to attempt (MAX_RETRY) (default %d)
+-p, --registry-password string   Sets the registry-password (REGISTRY_PASSWORD)
+-u, --registry-username string   Sets the registry username (REGISTRY_USERNAME)
+    --retry-interval duration    Retry interval (seconds) (RETRY_INTERVAL) (default %ds)
+    --timeout duration           Max duration (minutes) to wait for worker to complete (TIMEOUT) (default %dm)
+-v, --version                    Print version string
+`, defaultMaxFileSize, defaultMaxRetry, defaultRetryInterval, defaultTimeoutMinutes)
+
+	fmt.Print(usageHelp)
+}
+
+// parse and validate command line flags/env vars.
+func CollectCmdlineFlags(args []string) (CmdlineFlags, error) {
+	flags, err := parseCmdlineFlags(args)
+	if err != nil {
+		return flags, err
+	}
+	return flags, validateFlags(flags)
+}
+
+// parse flags from the command line and store them in the various fields of a CmdlineFlags struct.
+func parseCmdlineFlags(args []string) (CmdlineFlags, error) {
+	var flags CmdlineFlags
+
+	fs := flag.NewFlagSet("tink-worker", flag.ContinueOnError)
+	fs.Usage = Usage
+
+	// flags that have long and short options
+	fs.StringVar(&flags.DockerRegistry, "docker-registry", "", "")
+	fs.StringVar(&flags.DockerRegistry, "r", "", "")
+	fs.StringVar(&flags.RegistryUsername, "registry-username", "", "")
+	fs.StringVar(&flags.RegistryUsername, "u", "", "")
+	fs.StringVar(&flags.RegistryPassword, "registry-password", "", "")
+	fs.StringVar(&flags.RegistryPassword, "p", "", "")
+	fs.StringVar(&flags.WorkerID, "id", "", "")
+	fs.StringVar(&flags.WorkerID, "i", "", "")
+	fs.BoolVar(&flags.Version, "version", false, "")
+	fs.BoolVar(&flags.Version, "v", false, "")
+
+	// flags that are long-options only
+	fs.Int64Var(&flags.MaxFileSize, "max-file-size", defaultMaxFileSize, "")
+	fs.IntVar(&flags.MaxRetry, "max-retry", defaultMaxRetry, "")
+	fs.IntVar(&flags.RetryInterval, "retry-interval", defaultRetryInterval, "")
+	fs.IntVar(&flags.Timeout, "timeout", defaultTimeoutMinutes, "")
+
+	err := ff.Parse(fs, args, ff.WithEnvVarNoPrefix())
+	return flags, err
+}
+
+// perform command line flag validation.
+func validateFlags(flags CmdlineFlags) error {
+	var err error
+
+	// specifying the version flag will print the version and ignore everything else
+	if flags.Version {
+		fmt.Println(programVersion)
+		return err
```

Comment:
I apologize, I don't have a good reference but you might think about returning `nil` here instead of `err`. Just as you've done at the end of this function. I like to explicitly return nil when I know it should be. I've found it helpful when debugging. It allows me to be very confident that nil was returned instead of having to double-check `err` didn't get set unexpectedly.

Reply:
Updated as requested.